ktor-server/ktor-server-core/jvm/src/io/ktor/config/ApplicationConfig.kt

```diff
@@ -15,12 +20,12 @@ interface ApplicationConfig {
      * Get config property with [path] or fail
      * @throws ApplicationConfigurationException
      */
-    fun property(path: String): ApplicationConfigValue
+    fun property(path: String): HoconApplicationConfigValue
```

Comment:
Here we have abstraction so HOCON is just one of the implementations. We potentially can migrate to something else, for example, kotlinx.serialization. So we can't use this type here.

Reply:
Alright, I'll remove the deprecated `ApplicationConfig` typealias and reset these to it.